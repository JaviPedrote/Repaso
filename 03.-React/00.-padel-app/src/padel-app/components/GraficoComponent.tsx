import React from "react";
import { PieChart, Pie, Cell} from "recharts";

interface GraficoComponentProps {
  total: number;
  wins: number;
  losses: number;
}

export const GraficoComponent: React.FC<GraficoComponentProps> = ({ total, wins, losses }) => {
  const data = [
    { name: "Victorias", value: wins },
    { name: "Derrotas", value: losses },
  ];

  const COLORS = ["#ff6a6a", "#e51d1d"]; // Negro para ganados, Blanco para perdidos

  return (
    <div className="absolute ">
      <PieChart width={350} height={200}>
        <Pie
          data={data}
          cx="50%"
          cy="50%"
          innerRadius={60} // Hace el grÃ¡fico hueco
          outerRadius={80}
          fill="#8884d8"
          dataKey="value"
          label={({ name, value }) => `${name}: ${value}`}
          startAngle={90} endAngle={-270}
        >
          {data.map((entry, index) => (
            <Cell key={`cell-${index}`} fill={COLORS[index]} />
          ))}
        </Pie>
      </PieChart>
    </div>
  );
};
