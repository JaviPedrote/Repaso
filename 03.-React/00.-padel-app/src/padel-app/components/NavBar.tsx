
import { useDispatch } from 'react-redux'
import { logout } from '../../redux/slices/authSlice'
import { Link } from 'react-router-dom'


export const NavBar = () => {
    const dispatch = useDispatch()

    const onLogout = async () => {
        dispatch(logout())
        localStorage.removeItem('token')
    }
    return (
        <div className='bg-linear-to-br from-thunderbird-500 to-thunderbird-700 flex items-center justify-between opacity-85 z-50'>
            <h1 className="text-2xl font-bold text-white ml-8 py-2">Padel</h1>
            <div className='flex font-semibold text-white w-[60vw] justify-around'>
                <Link to={'/'}><h4 className='hover:scale-110 transition'>Home</h4></Link>
                <Link to={'/matches'}><h4 className='hover:scale-110 transition'>Matches</h4></Link>
                <Link to={'/ranking'}><h4 className='hover:scale-110 transition'>Ranking</h4></Link>
                <Link to={'/dasboard'}><h4 className='hover:scale-110 transition'>Dasboard</h4></Link>
            </div>
            <button className="px-3 py-0.5 bg-secondary text-white rounded mr-4 cursor-pointer hover:bg-texto hover:text-white transition-all delay-50 duration-300 hover:scale-120" onClick={onLogout}>Salir</button>
        </div>
    )
}
