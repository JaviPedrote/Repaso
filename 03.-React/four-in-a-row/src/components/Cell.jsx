

const Cell = ({ value, onClick }) => {
    return (
      <div 
        className="w-12 h-12 border-2 border-white flex justify-center items-center rounded-full bg-gray-500 cursor-pointer"
        onClick={onClick}
      >
        {value && (
          <div className={`w-10 h-10 rounded-full ${value === "red" ? "bg-red-500" : "bg-yellow-500"}`} />
        )}
      </div>
    );
  };
  
  export default Cell;
  