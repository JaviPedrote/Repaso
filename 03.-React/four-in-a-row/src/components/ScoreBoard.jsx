const ScoreBoard = ({ scores }) => {
    return (
      <div className="flex justify-between w-full p-4">
        <div className="text-white text-xl">ðŸ”´ Player 1: {scores.red}</div>
        <div className="text-white text-xl">ðŸŸ¡ Player 2: {scores.yellow}</div>
      </div>
    );
  };
  
  export default ScoreBoard;
  