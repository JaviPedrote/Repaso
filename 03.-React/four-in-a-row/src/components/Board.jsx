import { useState } from "react";
import Cell from "./Cell";

const ROWS = 6;
const COLS = 7;

const Board = ({ updateScore }) => {
  const [board, setBoard] = useState(Array(ROWS).fill(null).map(() => Array(COLS).fill(null)));
  const [currentPlayer, setCurrentPlayer] = useState("red");

  const dropDisc = (col) => {
    const newBoard = [...board];
    for (let row = ROWS - 1; row >= 0; row--) {
      if (!newBoard[row][col]) {
        newBoard[row][col] = currentPlayer;
        setBoard(newBoard);
        if (checkWin(newBoard, row, col, currentPlayer)) {
          updateScore(currentPlayer);
          alert(`${currentPlayer.toUpperCase()} Wins!`);
          resetGame();
        } else {
          setCurrentPlayer(currentPlayer === "red" ? "yellow" : "red");
        }
        break;
      }
    }
  };

  const checkWin = (board, row, col, player) => {
    const directions = [[1, 0], [0, 1], [1, 1], [1, -1]];
    for (let [dx, dy] of directions) {
      let count = 1;
      for (let dir = -1; dir <= 1; dir += 2) {
        let x = row + dir * dx;
        let y = col + dir * dy;
        while (x >= 0 && x < ROWS && y >= 0 && y < COLS && board[x][y] === player) {
          count++;
          x += dir * dx;
          y += dir * dy;
        }
      }
      if (count >= 4) return true;
    }
    return false;
  };

  const resetGame = () => {
    setBoard(Array(ROWS).fill(null).map(() => Array(COLS).fill(null)));
    setCurrentPlayer("red");
  };

  return (
    <div className="flex flex-col items-center">
      <div className="grid grid-cols-7 gap-2 p-4 bg-blue-500 rounded-lg">
        {board.map((row, rowIndex) =>
          row.map((cell, colIndex) => (
            <Cell key={`${rowIndex}-${colIndex}`} value={cell} onClick={() => dropDisc(colIndex)} />
          ))
        )}
      </div>
      <button onClick={resetGame} className="mt-4 bg-gray-700 text-white px-4 py-2 rounded">
        Restart
      </button>
    </div>
  );
};

export default Board;
