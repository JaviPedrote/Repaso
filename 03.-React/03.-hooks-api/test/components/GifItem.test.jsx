import { render, screen } from "@testing-library/react";

import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const titulo = "Pruebas en GifItem";

  const url = "https://localhost/algo.jpg";

  render(<GifItem title={titulo} url={url} />);

  test("debe mostrar el componente correctamente", () => {
    const { container } = render(<GifItem title={titulo} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostar la imagen con el url y el alt indicado", () => {
    render(<GifItem title={titulo} url={url} />);

    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(titulo);
  });

  test("debe mostar el titulo en el parrafo", () => {

    render(<GifItem title={titulo} url={url} />);
    
    expect(screen.getByText(titulo)).toBeTruthy();
  });
});
