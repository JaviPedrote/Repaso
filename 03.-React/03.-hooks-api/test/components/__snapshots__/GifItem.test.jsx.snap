// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem debe mostrar el componente correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Pruebas en GifItem"
      src="https://localhost/algo.jpg"
    />
    <p>
      Pruebas en GifItem
    </p>
  </div>
</div>
`;
